File: python/백준/다이나믹 프로그래밍/문자열/5525 IOIOI.py
N = int(input())
M = int(input())
S = input()

count = 0
pattern_length = 0

i = 0

while i < M - 2:
    if S[i:i+3] == "IOI":
        pattern_length += 1
        if pattern_length == N:
            count += 1
            pattern_length -= 1
        i += 2
    else:
        pattern_length = 0
        i += 1

print(count)

File: cpp/백준/문자열/5525 IOIOI.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, M;
    cin >> N >> M;
    string S;
    cin >> S;

    int count = 0;
    int pattern_length = 0;
    int i = 0;

    while (i < M - 2) {
        if (S.substr(i, 3) == "IOI") {
            pattern_length += 1;
            if (pattern_length == N) {
                count += 1;
                pattern_length -= 1;
            }
            i += 2;
        } else {
            pattern_length = 0;
            i += 1;
        }
    }

    cout << count << "\n";
    return 0;
}
